### PR TITLE
Added support for configuring 'Use Quay for Stage Containers Pushes' …

### PR DESCRIPTION
**Pull request by StanislavJochman**
Jira - 2320

### CODE REVIEW

File: library/errata_tool_product_version.py [hunk 1]
```diff
@@ -293,21 +290,20 @@ def run_module():
...
-        supports_check_mode=True
+        is_server_only=dict(type="bool", required=True),
+        enabled=dict(type="bool", default=True),
+        allow_rhn_debuginfo=dict(type="bool", required=True),
+        allow_buildroot_push=dict(type="bool", required=True),
+        is_oval_product=dict(type="bool", required=True),
+        is_rhel_addon=dict(type="bool", required=True),
+        push_targets=dict(type="list", required=True),
+        brew_tags=dict(type="list", required=True),
+        use_quay_for_containers=dict(type="bool", default=False),
```
Comment (ktdreyer) [inline library/errata_tool_product_version.py:305]: Please take out the `default` values (`default=False`). This is a pattern that I wish I had not used early on here. It means that if a playbook author omits the `use_quay_for_containers` parameter on an `errata_tool_product_version` task, Ansible will surprisingly reset the value of `use_quay_for_containers` to `False` on the server, even if it was set to `True` on the server previously.

Instead, you can let it simply default to `None`. Later on, right before you call `ensure_product_version()`, if the value is indeed `None`, then drop it from `params`. Like this:

```python
    # `use_quay_for_containers` is optional - don't modify the server-side value if the user has omitted it.
    if params['use_quay_for_containers'] is None:
        params.pop('use_quay_for_containers')
```

File: library/errata_tool_product_version.py [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
-    'status': ['preview'],
-    'supported_by': 'community'
+    "metadata_version": "1.0",
+    "status": ["preview"],
+    "supported_by": "community",
 }
 
 
-DOCUMENTATION = '''
+DOCUMENTATION = """
```
Comment (ktdreyer) [inline library/errata_tool_product_version.py:13]: Please read the earlier PR in https://github.com/ktdreyer/errata-tool-ansible/pull/264 . We need to have new documentation for any parameters we add.

**Review by ktdreyer — CHANGES_REQUESTED** [review]
Thank you for this PR!

Would you please take out all the stylistic changes (single-quote to double-quotes, for example)? (I wonder if you ran `black` on this code, or if your editor did it automatically.) This PR should do [one thing](https://fagnerbrack.com/one-pull-request-one-concern-e84a27dfe9f1).

I'd rather not use `black` at this point. I've found that Stickler CI does a decent job cleaning up obvious flake8 errors, and the single-double-quotes thing in particular is just more typing and more noise on the screen to read.

Please also read the previous PR that does this in https://github.com/ktdreyer/errata-tool-ansible/pull/264 . @goyalmayankk added inline documentation and edited the unit tests, and we need to do the same in this PR.

File: library/errata_tool_product_version.py (hunk 2 above)
Comment (StanislavJochman) [reply library/errata_tool_product_version.py:13]: Hi I am using autoformater and did not know that would be a problem will fix that. 

Best Regards 
Stanislav Jochman

**Comment by StanislavJochman** [comment]
@ktdreyer I fixed problems in last PR. Could you take a look?

**Pull request closed without merging**